tests/PHPStan/Analyser/data/bug-8084.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Bug8084;
+
+use function array_shift;
+use function PHPStan\Testing\assertType;
+
+class Bug8084
+{
+	public function run(array $arr): void
+	{
+		/** @var array{a: 0, b?: 1} $arr */
```

Comment:
Hi, please put this type into `@param` above the method, here it's ambiguous what it means.

Also currently this example https://phpstan.org/r/de642079-e5b0-492a-9782-d78c324f70a7 shows `Expression on left side of ?? is not nullable.` so we'll need another test in NullCoalesceRuleTest to make sure this message goes away.

Thank you!

Reply:
Hi, thank you for your review.
Added a test, but included a merge commit to follow the latest branch.  
Should I recreate?